### PR TITLE
[Heartbeat] Added synthetics step status field

### PR DESCRIPTION
**Pull request by shahzad31**
## What does this PR do?

Resolves: https://github.com/elastic/beats/issues/24141

Added `synthetics.step.status` field and relevant mapping, status originally exists in `synthetics.payload.status`.

## How to test this PR locally

Ran an example journey with heartbeat and it pushed the status succesfully 

```
          "synthetics" : {
            "index" : 2,
            "step" : {
              "name" : "load homepage",
              "index" : 1,
              "status" : "succeeded"
            },
            "journey" : {
              "id" : "inline",
              "name" : "inline"
            },
            "type" : "step/end",
            "package_version" : "0.0.1-alpha.9",
            "payload" : {
              "url" : "https://www.elastic.co/",
              "status" : "succeeded",
              "source" : "async () => { await page.goto('https://www.elastic.co'); }",
              "start" : 17182.069648224,
              "end" : 17188.741199725
            }
          },
```

## Screenshots

image

### CODE REVIEW

File: x-pack/heartbeat/monitors/browser/synthexec/synthexec.go [hunk 1]
```diff
@@ -244,6 +244,13 @@ func jsonToSynthEvent(bytes []byte, text string) (res *SynthEvent, err error) {
 
 	res = &SynthEvent{}
 	err = json.Unmarshal(bytes, res)
+	stepStatus, _ := res.Payload.GetValue("status")
```
Comment (andrewvc) [inline x-pack/heartbeat/monitors/browser/synthexec/synthexec.go:247]: I'm wondering if we should make this change in `@elastic/synthetics` so it's in a more strongly typed field than `payload`. @vigneshshanmugam @shahzad31 WDYT?

Comment (shahzad31) [reply x-pack/heartbeat/monitors/browser/synthexec/synthexec.go:247]: i do agree, moving it to synthetics will make heartbeat code clean

Comment (vigneshshanmugam) [reply x-pack/heartbeat/monitors/browser/synthexec/synthexec.go:247]: ++, We should move both the journey and step status out of the payload fields and have a proper schema.

Comment (vigneshshanmugam) [reply x-pack/heartbeat/monitors/browser/synthexec/synthexec.go:247]: Created an issue - https://github.com/elastic/synthetics/issues/230

**Comment by shahzad31** [comment]
With https://github.com/elastic/synthetics/pull/231 merged ,i have simplified the status field for step.

File: heartbeat/_meta/fields.common.yml [hunk 2]
```diff
@@ -156,6 +156,8 @@
               type: text
             - name: index
               type: integer
+            - name: status
+              type: text
```
Comment (andrewvc) [inline heartbeat/_meta/fields.common.yml:160]: This should be keyword, not text

**Review by andrewvc — CHANGES_REQUESTED** [review]
Works great except the one mapping issue, change that and we're LGTM

File: x-pack/heartbeat/monitors/browser/synthexec/execmultiplexer_test.go [hunk 3]
```diff
@@ -33,8 +33,9 @@ func TestExecMultiplexer(t *testing.T) {
 
 		for sIdx := 0; sIdx < 3; sIdx++ {
 			step := &Step{
-				Name:  fmt.Sprintf("S%d", sIdx),
-				Index: sIdx,
+				Name:   fmt.Sprintf("S%d", sIdx),
+				Index:  sIdx,
+				Status: fmt.Sprintf("failed%d", sIdx),
```
Comment (andrewvc) [inline x-pack/heartbeat/monitors/browser/synthexec/execmultiplexer_test.go:38]: Why do we need to interpolate the step number into the status? The status field is an enum. The reason we interpolate the index into the step name is to give unique step names, but that doesn't apply here.

File: heartbeat/_meta/fields.common.yml (hunk 2 above)
Comment (shahzad31) [reply heartbeat/_meta/fields.common.yml]: fixed !!

File: x-pack/heartbeat/monitors/browser/synthexec/execmultiplexer_test.go (hunk 3 above)
Comment (shahzad31) [reply x-pack/heartbeat/monitors/browser/synthexec/execmultiplexer_test.go]: fixed !!

**Comment by shahzad31** [comment]
@elasticmachine merge upstream

**Comment by shahzad31** [comment]
@andrewvc CI is green finally

**Review by andrewvc — APPROVED** [review]
LGTM

**Pull request merged**